[Model Monitoring] Remove TDEngine TSDB support

alxtkr77 opened the pull request:
### Description
Remove all TDEngine TSDB support from MLRun codebase. TimescaleDB (PostgreSQL) remains as the primary TSDB backend alongside V3IO TSDB.

---

### Changes Made
- Delete `mlrun/model_monitoring/db/tsdb/tdengine/` directory 
- Remove `TDEngineConnector`, `TDEngineConnection`, and related schemas
- Remove `DatastoreProfileTDEngine` from datastore profiles
- Remove `TDEngineStoreyTarget` from storey targets
- Remove `TSDBTarget.TDEngine` and `TDEngineSuperTables` enums
- Remove `taos-ws-py==0.3.2` from all requirements files
- Delete `tests/model_monitoring/db/tsdb/tdengine/` test directory
- Delete `tests/model_monitoring/test_tdengine.py`
- Remove TDEngine from parametrized tests
- Remove `TestTDEngineProfile` test class
- Update model monitoring documentation to reference TimescaleDB
- Update installation guides (kubernetes.md, aws-install.md)
- Update ecosystem.md and tutorials

---

### Checklist
- [x] I updated the documentation (if applicable)
- [x] I have tested the changes in this PR

---

### Testing
- Ran `make fmt` - passed
- Ran `make lint` - all checks passed
- Verified no TDEngine/taos references remain in Python, YAML, or notebook files
- Historical deprecation notices in change-log are expected and remain

---

### References
- Ticket link: [ML-11628](https://iguazio.atlassian.net/browse/ML-11628)

---

### Breaking Changes?

- [x] Yes (explain below)
- [ ] No

TDEngine TSDB support is removed. Users who were using TDEngine for model monitoring must migrate to TimescaleDB (PostgreSQL) or V3IO TSDB



[ML-11628]: https://iguazio.atlassian.net/browse/ML-11628?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

assaf758 commented:
Fails in test:
```
FAILED tests/integration/sdk_api/hub/test_hub.py::TestHub::test_import_module_from_hub - mlrun.errors.MLRunBadRequestError: Failed to import module from temp/agent_deployer.py: cannot import name 'DatastoreProfileTDEngine' from 'mlrun.datastore.datastore_profile' (/mlrun/mlrun/datastore/datastore_profile.py)
```
Seems you need to update https://github.com/mlrun/functions/tree/master/modules/src/agent_deployer

Commit:
[Model Monitoring] Remove TDEngine TSDB support

Remove all TDEngine TSDB support from MLRun codebase. TDEngine was
deprecated in v1.10.0 following the 2-release deprecation cycle.
TimescaleDB (PostgreSQL) remains as the primary TSDB backend alongside
V3IO TSDB.

Core changes:
- Delete mlrun/model_monitoring/db/tsdb/tdengine/ directory (6 files)
- Remove TDEngineConnector, TDEngineConnection, and related schemas
- Remove DatastoreProfileTDEngine from datastore profiles
- Remove TDEngineStoreyTarget from storey targets
- Remove TSDBTarget.TDEngine and TDEngineSuperTables enums

Dependencies:
- Remove taos-ws-py==0.3.2 from all requirements files
- Update extras-requirements.txt and locked requirements

Tests:
- Delete tests/model_monitoring/db/tsdb/tdengine/ directory
- Delete tests/model_monitoring/test_tdengine.py
- Remove TDEngine from parametrized tests
- Remove TestTDEngineProfile test class

Reference: ML-11628

Commit:
[Model Monitoring] Update documentation for TDEngine removal

Update model monitoring docs to reference TimescaleDB instead of TDEngine.

Changes:
- Update model monitoring docs and installation guides
- Update kubernetes.md and aws-install.md
- Update ecosystem.md to list TimescaleDB instead of TDEngine
- Update tutorials and Jupyter notebooks

Reference: ML-11628
